Sources/Tracking/ModuleTopLevelObject.swift

```diff
@@ -14,17 +14,21 @@ open class ModuleTopLevelObject<ImplementationClass> {
         alreadyCreatedImplementation ?? createAndSetImplementationInstance()
     }
 
+    public var globalDataStore: GlobalDataStore
+
     private let sdkInitializedUtil: SdkInitializedUtil
 
     // for writing tests
     // provide a nil implementation if you want `sdkInitializedUtil` logic to run and real instance of implementation to run in tests
-    public init(implementation: ImplementationClass?, sdkInitializedUtil: SdkInitializedUtil) {
+    public init(implementation: ImplementationClass?, globalDataStore: GlobalDataStore, sdkInitializedUtil: SdkInitializedUtil) {
```

Comment:
Does `globalDataStore` need to be part of this `init` function and class? 

Since I don't see this class using this dependency, I would prefer we don't keep an instance of it in the class.

Reply:
`MessagingPush` is a singleton class. So to make it injectable, and testable I think this was the ideal way. But I might not be fully integrated with the iOS structure. So if there is any other way I can make it injectable in `MessagingPush` class?

Reply:
I believe if you modify [the constructor of `MessagingPush`](https://github.com/customerio/customerio-ios/blob/3dbc9bd9e68c8fab57d8c13578f3be1d89387c17/Sources/MessagingPush/MessagingPush.swift#L13-L14) to something like this:

```swift
public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, MessagingPushInstance {
    @Atomic public private(set) static var shared = MessagingPush()
	private let globalDataStore: GlobalDataStore

    // testing constructor
    override internal init(implementation: MessagingPushInstance?, globalDataStore: GlobalDataStore, sdkInitializedUtil: SdkInitializedUtil) {
        super.init(implementation: implementation, sdkInitializedUtil: sdkInitializedUtil)
		self.globalDataStore = globalDataStore
    }
```

Then you wouldn't need to include `globalDataStore` in this `ModuleTopLevelObject` class.

Reply:
I don't think that will work, a subclass constructor cannot have more parameters than its superclass constructor.
It will return `Initializer does not override a designated initializer from its superclass`

Reply:
I am trying something out in Xcode right now. I'll update this comment or push a commit if I find something that will work.

Reply:
[Found a solution](https://github.com/customerio/customerio-ios/pull/285/commits/01f66a437a1596e10e45cce78e3a4d517dee0204#diff-d322168af6d12b45ff1a57021d6a686396fe280ff72e00f2c1872bc9e4abff3fR11-R21). 

As always, we can revert this commit if we don't like the solution. 

Otherwise, let's resolve this conversation.